aptos-move/block-executor/src/txn_commit_listener.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    errors::Error,
+    task::{ExecutionStatus, TransactionOutput},
+};
+use aptos_mvhashmap::types::TxnIndex;
+use std::fmt::Debug;
+
+/// An interface for listening to transaction commit events. The listener is called only once
+/// for each transaction commit.
+pub trait TransactionCommitListener: Send + Sync {
+    type ExecutionStatus;
+
+    fn on_transaction_committed(&self, txn_idx: TxnIndex, execution_status: &Self::ExecutionStatus);
```

Comment:
i'm not entirely sure whether we need this index, but anyway I think we may push this `on_commit` function to the trait TransactionOutput